core/src/main/java/org/apache/accumulo/core/data/ResourceGroupId.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import org.apache.accumulo.core.Constants;
+import org.apache.accumulo.core.util.cache.Caches;
+import org.apache.accumulo.core.util.cache.Caches.CacheName;
+
+import com.github.benmanes.caffeine.cache.Cache;
+
+public class ResourceGroupId extends AbstractId<ResourceGroupId> {
+
+  // cache is for canonicalization/deduplication of created objects,
+  // to limit the number of TableId objects in the JVM at any given moment
+  // WeakReferences are used because we don't need them to stick around any longer than they need to
+  static final Cache<String,ResourceGroupId> cache = Caches.getInstance()
+      .createNewBuilder(CacheName.RESOURCE_GROUP_ID, false).weakValues().build();
+
+  public static final ResourceGroupId DEFAULT =
+      ResourceGroupId.of(Constants.DEFAULT_RESOURCE_GROUP_NAME);
+
+  private static final long serialVersionUID = 1L;
+
+  private ResourceGroupId(String canonical) {
+    super(canonical);
```

Comment:
Noticed we can create ids using empty string.  Seem like it could be troublesome to have a RG that is empty string.  Wonder if we should error for that case (maybe blank too).  Also wondering if other ids should have that restriction.